software/control_sw/src/cosmic_fengine.py

```diff
@@ -1437,36 +1447,54 @@ def delay_tracking(self):
 
             else:
                 #No values received, load values on hand
-                required_loadtime_s = int(delay_coeffs['loadtime']) #assuming in integer seconds
-                required_loadtime_fpga_clks = required_loadtime_s * FPGA_CLOCK_RATE_HZ
-                
+                t = time.time() * 1e6 # received loadtime is in microseconds so compare against time now in microseconds
+
+                if (delay_coeffs['loadtime_us'] < (t + 1e5)
+                    or 
+                    delay_coeffs['loadtime_us'] > (t + 1e6)):
+                    #if the provided loadtime is not at least 1e-1s into the future or more than 1s into the future
+                    required_loadtime_us = t + 1e6 # interpolate one second into the future on the old delay coefficients
+                    required_loadtime_s = required_loadtime_us * 1e-6
+                else:
+                    #if the provided loadtime is in the future but within 1 second 
```

Comment:
Suggest a log here that new targets are being loaded

Reply:
Ace, will do.